File: app/models/entities/Event.go
package entities

import "time"

// Event структура сущности события
type Event struct {
	ID       int64      `json:"ID"`       // идентификатор
	Book     *Book      `json:"book"`     // книга, связанная с событием
	Employee *Employee  `json:"employee"` // сотрудник, связанный с событием
	Type     string     `json:"type"`     // тип события
	Date     *time.Time `json:"date"`     // дата события
}

// EventType тип события
type EventType string

const (
	EVENT_TYPE_GIVE EventType = "выдача"
	EVENT_TYPE_TAKE EventType = "сдача"
)

File: bin/linux/src/sample-project/public/js/src/models/entities/event.js
// Event класс для представления сущности события выдачи/сдачи книги
export class Event {
    constructor(id, book, employee, type, date) {
        this.ID = id
        this.book = book
        this.employee = employee
        this.type = type
        this.date = date
    }
}

// допустимые типы события
export let EVENT_TYPE = {
    give: 'выдача',
    take: 'сдача',
}